x-pack/legacy/plugins/ml/server/routes/filters.ts

```diff
@@ -0,0 +1,227 @@
+  const mgr = new FilterManager(context.ml!.mlClient.callAsCurrentUser);
+  return mgr.getAllFilterStats();
+}
+
+function getFilter(context: RequestHandlerContext, filterId: string) {
+  const mgr = new FilterManager(context.ml!.mlClient.callAsCurrentUser);
+  return mgr.getFilter(filterId);
+}
+
+function newFilter(context: RequestHandlerContext, filter: FormFilter) {
+  const mgr = new FilterManager(context.ml!.mlClient.callAsCurrentUser);
+  return mgr.newFilter(filter);
+}
+
+function updateFilter(context: RequestHandlerContext, filterId: string, filter: FormFilter) {
+  const mgr = new FilterManager(context.ml!.mlClient.callAsCurrentUser);
+  return mgr.updateFilter(filterId, filter);
+}
+
+function deleteFilter(context: RequestHandlerContext, filterId: string) {
+  const mgr = new FilterManager(context.ml!.mlClient.callAsCurrentUser);
+  return mgr.deleteFilter(filterId);
+}
+
+export function filtersRoutes({ xpackMainPlugin, router }: RouteInitialization) {
+  /**
+   * @apiGroup Filters
+   *
+   * @api {get} /api/ml/filters Gets filters
+   * @apiName GetFilters
+   * @apiDescription Retrieves the list of filters which are used for custom rules in anomaly detection.
+   *
+   * @apiSuccess {Boolean} success
+   * @apiSuccess {Object} filters list of filters
+   */
+  router.get(
+    {
+      path: '/api/ml/filters',
+      validate: false,
+    },
+    licensePreRoutingFactory(xpackMainPlugin, async (context, request, response) => {
+      try {
+        const resp = await getAllFilters(context);
+
+        return response.ok({
+          body: resp,
+        });
+      } catch (e) {
+        return response.customError(wrapError(e));
+      }
+    })
+  );
+
+  /**
+   * @apiGroup Filters
+   *
+   * @api {get} /api/ml/filters/:filterId Gets filter by ID
+   * @apiName GetFilterById
+   * @apiDescription Retrieves the filter with the specified ID.
+   *
+   * @apiSuccess {Boolean} success
+   * @apiSuccess {Object} filter the filter with the specified ID
+   */
+  router.get(
+    {
+      path: '/api/ml/filters/{filterId}',
+      validate: {
+        params: schema.object({ filterId: schema.string() }),
+      },
+    },
+    licensePreRoutingFactory(xpackMainPlugin, async (context, request, response) => {
+      try {
+        const resp = await getFilter(context, request.params.filterId);
+        return response.ok({
+          body: resp,
+        });
+      } catch (e) {
+        return response.customError(wrapError(e));
+      }
+    })
+  );
+
+  /**
+   * @apiGroup Filters
+   *
+   * @api {put} /api/ml/filters Creates a filter
+   * @apiName CreateFilter
+   * @apiDescription Instantiates a filter, for use by custom rules in anomaly detection.
+   *
+   * @apiSuccess {Boolean} success
+   * @apiSuccess {Object} filter created filter
+   */
+  router.put(
+    {
+      path: '/api/ml/filters',
+      validate: {
+        body: schema.object({ ...createFilterSchema }),
```

Comment:
We don't actually need to create the new object here with the spread operator - it can just be passed in plainly like `schema.object(createFilterSchema)`. 
I did it this way by mistake the first time and will clean those up. It still works but is just unnecessary.

Reply:
Done in https://github.com/elastic/kibana/pull/57597/commits/8cd2ec85d391ca81e15f50b70a9531e01891fe68